src/main/java/com/google/netpcapanalysis/dao/MaliciousIPDaoImpl.java

```diff
@@ -73,4 +78,65 @@ else if (result.getBody().contains(REQUEST_LIMIT))
         return data;
     }
     
+    private String isMalicious(String ip)
+    {
+        HttpResponse<String> apiResult;
+        String result;
+
+        String searchDB = ipCache.searchMaliciousDB(ip);
+        if(searchDB.equalsIgnoreCase(Flagged.TRUE))
+        {
+            result = Flagged.TRUE;
+        }
+        else if(searchDB.equalsIgnoreCase(Flagged.FALSE)){
+            result = Flagged.FALSE;
+        }
+        else{
+            System.out.println("Else statement");
+            try {
+                apiResult = Unirest.get("https://signals.api.auth0.com/badip/" + ip)
+                        .header("X-Auth-Token", AUTH0_API_KEY)
+                        .asString();
+           
+                if(apiResult.getBody().equalsIgnoreCase(FLAGGED_FALSE))
+                {
+                    result = Flagged.FALSE;
+                    MaliciousPacket tempPacket = new MaliciousPacket(ip, result);
+                    ipCache.setMaliciousIPObjects(tempPacket);
+                }
+                else if(apiResult.getBody().equalsIgnoreCase(FLAGGED_TRUE))
+                {
+                    result = Flagged.TRUE; 
+                    MaliciousPacket tempPacket = new MaliciousPacket(ip, result);
+                    ipCache.setMaliciousIPObjects(tempPacket);
+                }
+                else if (apiResult.getBody().contains(REQUEST_LIMIT))
+                {
+                    result = Flagged.UNKNOWN; 
+                }
+                else{
+                    result = Flagged.ERROR; 
+                }
+    
+            } catch (UnirestException e) {
+                System.out.println("IP lookup failed");
+                result = Flagged.ERROR;
+                // TODO Auto-generated catch block
+                e.printStackTrace();
+            }
+        }
+        return result;
+    }
+    
+    public void run(String searchEntity){
+        PCAPDao datastore = new PCAPDaoImpl();
+        ArrayList<PCAPdata> allData = datastore.getPCAPObjects(searchEntity); 
+        ArrayList<PCAPdata> uniqueData = datastore.getUniqueIPs(allData); 
```

Comment:
The datastore should not provide you the uniqueIPs. It is a utility independent of that datastore. It just tkaes in a list of PCAP data and returns a List of IPs.